api/turing/api/pod_log_api.go

```diff
@@ -48,78 +68,217 @@ func (c PodLogController) ListPodLogs(_ *http.Request, vars RequestVars, _ inter
+
+	request := service.PodLogRequest{
+		Namespace:        servicebuilder.GetNamespace(project),
+		DefaultContainer: cluster.KnativeUserContainerName,
+		Environment:      router.EnvironmentName,
+		LabelSelectors: []service.LabelSelector{
+			{
+				Key:   cluster.KnativeServiceLabelKey,
+				Value: servicebuilder.GetComponentName(routerVersion, componentType),
+			},
+		},
+	}
+	varLogError := c.parsePodLogRequest(&request, vars)
+
+	if varLogError != nil {
+		return BadRequest(varLogError.Description, varLogError.ErrorMessage)
+	}
+
+	logs, err := c.PodLogService.ListPodLogs(request)
+	if err != nil {
+		return InternalServerError("Failed to list logs", err.Error())
+	}
+
+	return Ok(logs)
+}
+
+// ListEnsemblingJobPodLogs handles the HTTP request for getting Ensembling Pod Logs
+// It supports 3 types of pods:
+//   1. Image Builder
+//   2. Spark Driver
+//   3. Spark Executor
+func (c PodLogController) ListEnsemblingJobPodLogs(_ *http.Request, vars RequestVars, _ interface{}) *Response {
+	var errResp *Response
+	var project *mlp.Project
+	if project, errResp = c.getProjectFromRequestVars(vars); errResp != nil {
+		return errResp
+	}
+
+	componentType, err := getComponentType(batch.ImageBuilderPodType, vars, batchEnsemblingAllowedComponents)
+	if err != nil {
+		return BadRequest(err.Error(), fmt.Sprintf(
+			"must be one of the following: %s, %s or %s",
+			batch.ImageBuilderPodType,
+			batch.DriverPodType,
+			batch.ExecutorPodType,
+		))
 	}
 
-	opts := &service.PodLogOptions{}
-	opts.Container, _ = vars.get("container")
+	options := &ListEnsemblingPodLogsOptions{}
+	if err := c.ParseVars(options, vars); err != nil {
+		return BadRequest(
+			"failed to fetch ensembling job",
+			fmt.Sprintf("failed to parse query string: %s", err),
+		)
+	}
+
+	ensemblingJob, err := c.EnsemblingJobService.FindByID(
+		*options.ID,
+		service.EnsemblingJobFindByIDOptions{
+			ProjectID: options.ProjectID,
+		},
+	)
+
+	if err != nil {
+		return NotFound("ensembling job not found", err.Error())
+	}
+
+	namespace := c.EnsemblingJobService.GetNamespaceByComponent(componentType, project)
+	environment := c.EnsemblingJobService.GetDefaultEnvironment()
+	ensemblerName := ensemblingJob.InfraConfig.EnsemblerName
+	request := service.PodLogRequest{
+		Namespace:   namespace,
+		Environment: environment,
+		LabelSelectors: c.EnsemblingJobService.CreatePodLabelSelector(
+			ensemblerName,
+			componentType,
+		),
+	}
+
+	varLogError := c.parsePodLogRequest(&request, vars)
+	if varLogError != nil {
+		return BadRequest(varLogError.Description, varLogError.ErrorMessage)
+	}
+
+	legacyFormatLogs, err := c.PodLogService.ListPodLogs(request)
+	if err != nil {
+		return InternalServerError("Failed to list logs", err.Error())
+	}
+
+	loggingURL, err := c.EnsemblingJobService.FormatLoggingURL(ensemblerName, namespace, componentType)
+	if err != nil {
+		return InternalServerError("Failed to format monitoring URL", err.Error())
+	}
+
+	logs := service.ConvertPodLogsToV2(
```

Comment:
Maybe add some inline comments here about this logic?

Reply:
alright